app/Models/Account.php

```
@@ -63,8 +74,15 @@ public function getFullAccountAttribute(): string
         return "{$this->account_number} - {$this->account_name} $reportGroup";
     }
 
-    public function scopeWithAccountName($query, $accountName)
+    public function scopeAccountName(Builder $query, string $accountName): Builder
     {
-        return $query->where('account_name', $accountName)->first()->id;
+        return $query->where('account_name', $accountName);
     }
+
+    public function scopeTaxable(Builder $query, bool $state = true): Builder
```

PHPMD (2.15.0)

The method scopeTaxable has a boolean flag argument $state, which is a certain sign of a Single Responsibility Principle violation. (Clean Code Rules)

(BooleanArgumentFlag)